src/chrony.py

```diff
@@ -0,0 +1,406 @@
+# Copyright 2025 Canonical Ltd.
+# See LICENSE file for licensing details.
+
+"""Chrony controller."""
+
+# check chrony.conf document for _PoolOptions attributes.
+# flake8: noqa: DCO060
+
+import collections
+import itertools
+import logging
+import pathlib
+import shutil
+import subprocess  # nosec
+import textwrap
+import typing
+import urllib.parse
+
+import pydantic
+from charms.operator_libs_linux.v0 import apt
+from charms.operator_libs_linux.v1 import systemd
+
+logger = logging.getLogger(__name__)
+
+
+class _PoolOptions(pydantic.BaseModel):
+    """Chrony pool directive options.
+
+    For more detail: https://chrony-project.org/doc/4.5/chrony.conf.html
+    """
+
+    model_config = pydantic.ConfigDict(extra="forbid")
+
+    minpoll: int | None = None
+    maxpoll: int | None = None
+    iburst: bool = False
+    burst: bool = False
+    key: str | None = None
+    nts: bool = False
+    certset: str | None = None
+    maxdelay: float | None = None
+    maxdelayratio: float | None = None
+    maxdelaydevratio: float | None = None
+    maxdelayquant: float | None = None
+    mindelay: float | None = None
+    asymmetry: float | None = None
+    offset: float | None = None
+    minsamples: int | None = None
+    maxsamples: int | None = None
+    filter: int | None = None
+    offline: bool = False
+    auto_offline: bool = False
+    prefer: bool = False
+    noselect: bool = False
+    trust: bool = False
+    require: bool = False
+    xleave: bool = False
+    polltarget: int | None = None
+    presend: int | None = None
+    minstratum: int | None = None
+    version: int | None = None
+    extfield: str | None = None
+    maxsources: int | None = None
+
+    def render_options(self) -> str:
+        """Render pool options as chrony option string.
+
+        Returns:
+            Chrony pool directive option string.
+        """
+        options = []
+        # mypy and pylint have problems handling the model_fields class attribute.
+        # pylint: disable=not-an-iterable
+        for field in sorted(f for f in _PoolOptions.model_fields if f != "copy"):  # type: ignore
+            value = getattr(self, field)
+            if value is True:
+                options.append(field)
+            elif value is None or value is False:
+                continue
+            else:
+                options.extend([field, str(value)])
```

Comment:
Add a comment that:
- First we look for boolean which is True
- Then we look for value which is None or boolean which is False
- And finally for integer or strings

Reply:
Added, thanks!